onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -13,7 +15,209 @@
 #include "onnx/defs/tensor_proto_util.h"
 #include "core/mlas/inc/mlas.h"
 
+template<const char* const *... args>
+struct StaticStrings {
+    static constexpr const char * const data[] = {*args...};
+};
+
+constexpr auto TENSOR_INT8 = "tensor(int8)";
+constexpr auto TENSOR_UINT8 = "tensor(uint8)";
+constexpr auto TENSOR_INT16 = "tensor(int16)";
+constexpr auto TENSOR_UINT16 = "tensor(uint16)";
+constexpr auto TENSOR_INT32 = "tensor(int32)";
+constexpr auto TENSOR_UINT32 = "tensor(int32)";
+constexpr auto TENSOR_INT64 = "tensor(int64)";
+constexpr auto TENSOR_UINT64 = "tensor(int64)";
+constexpr auto TENSOR_FLOAT = "tensor(float)";
+constexpr auto TENSOR_FLOAT16 = "tensor(float16)";
+constexpr auto TENSOR_DOUBLE = "tensor(double)";
+constexpr auto TENSOR_STRING = "tensor(string)";
+constexpr auto TENSOR_BOOL = "tensor(bool)";
+constexpr auto TENSOR_COMPLEX64 = "tensor(complex64)";
+constexpr auto TENSOR_COMPLEX128 = "tensor(complex128)";
+
 namespace ONNX_NAMESPACE {
+
+template<typename U>
+struct span {
+    class iterator {
+    public:
+        iterator(const U *ptr): ptr(ptr) {}
+        iterator operator++() { ++ptr; return *this; }
+        bool operator!=(const iterator& other) const { return ptr != other.ptr; }
+        const U& operator*() const { return *ptr; }
+    private:
+        U* ptr;
+    };
+
+    const U *data;
+    size_t n;
+
+    constexpr span() : data(), n() {}
+
+    template <size_t N>
+    constexpr span(const U (&arr)[N])
+    : data(arr), n(N) {}
+
+    iterator begin() const { return iterator(data); }
+    iterator end() const { return iterator(data + n); }
+    size_t size() const { return n; }
+    const U& operator[](size_t i) const { return data[i]; }
+};
```

Comment:
I'm using a custom span class to work around this compiler bug: https://developercommunity.visualstudio.com/content/problem/1110064/valid-pointer-arithmetic-rejected-in-constexpr-con.html